packages/new-deploy/src/DeployContext.ts

```diff
@@ -0,0 +1,298 @@
+  }
+
+  getCore(nameOrDomain: string | number): CoreContracts | undefined {
+    const name = this.resolveDomainName(nameOrDomain);
+    if (!this.data.core[name]) return undefined;
+    return new CoreContracts(this, name, this.data.core[name]);
+  }
+
+  mustGetCore(nameOrDomain: string | number): CoreContracts {
+    const core = this.getCore(nameOrDomain);
+    if (!core) throw new Error(`No core contracts for domain ${nameOrDomain}`);
+    return core;
+  }
+
+  getBridge(nameOrDomain: string | number): BridgeContracts | undefined {
+    const name = this.resolveDomainName(nameOrDomain);
+    if (!this.data.bridge[name]) return undefined;
+    return new BridgeContracts(this, name, this.data.bridge[name]);
+  }
+
+  mustGetBridge(nameOrDomain: string | number): BridgeContracts {
+    const bridge = this.getBridge(nameOrDomain);
+    if (!bridge)
+      throw new Error(`No bridge contracts for domain ${nameOrDomain}`);
+    return bridge;
+  }
+
+  addDomain(domain: config.Domain): void {
+    this._data = config.addNetwork(this.data, domain);
+  }
+
+  protected addCore(name: string, core: config.EvmCoreContracts): void {
+    this._data = config.addCore(this.data, name, core);
+  }
+
+  protected addBridge(name: string, bridge: config.EvmBridgeContracts): void {
+    this._data = config.addBridge(this.data, name, bridge);
+  }
+
+  pushVerification(
+    nameOrDomain: string | number,
+    verification: Verification,
+  ): void {
+    const name = this.resolveDomainName(nameOrDomain);
+    if (!this.verification.has(name)) this.verification.set(name, []);
+
+    const net = this._verification.get(name);
+    if (net) net.push(verification);
+  }
+
+  protected async deployCore(domain: config.Domain): Promise<void> {
+    this.addDomain(domain);
+
+    const core = new CoreContracts(this, domain.name);
+    await core.recordStartBlock();
+
+    await Promise.all([
+      core.deployUpgradeBeaconController(),
+      core.deployUpdaterManager(),
+      core.deployXAppConnectionManager(),
+    ]);
+
+    await Promise.all([core.deployHome(), core.deployGovernanceRouter()]);
+
+    // all contracts deployed
+    const complete = core.complete();
+    this.addCore(domain.name, complete);
+
+    // the set of domains that are not the new core
+    const remoteDomains = this.data.networks.filter(
+      (net) => net !== domain.name,
+    );
+
+    // post-deployment setup
+    // We choose to enroll all currently known routers and watchers here,
+    // event those that may not be immediately connected, so that IF they are
+    // connected in the future, the routers will be properly configured.
+    // Essentially we run likely future governance actions at deploy time, so
+    // we will not be required to run them later.
+    await Promise.all([
+      core.xAppConnectionManager.setHome(complete.home.proxy),
+      core.updaterManager.setHome(complete.home.proxy),
+      ...remoteDomains.map((remote) => core.enrollGovernanceRouter(remote)),
```

Comment:
If we deploy to 2 networks, it will happen that the remote won't be ready by the time the first one will reach this line